html/pfappserver/lib/pfappserver/PacketFence/Controller/Config/Switch.pm

```diff
@@ -274,6 +275,73 @@ sub invalidate_cache :Chained('object') :PathPart('invalidate_cache') :Args(0) {
     $c->response->status(200);
 }
 
+=head1 import_csv
+
+A method to be able to import switches from a CSV
+
+=cut
+
+sub import_csv :Local :Args(0) :AdminRole('SWITCHES_CREATE') {
+    my ( $self, $c ) = @_;
+   
+    my $logger = pf::log->get_logger();
+    my $upload = $c->req->upload('importcsv');
+    my $file = $upload->fh;
+    my $model = $c->model("Config::Switch");
+    my $model_group = $c->model("Config::SwitchGroup");
```

Comment:
It would be better to use Text::CSV
To take care of parsing quoted and new lines.
Look at html/pfappserver/lib/pfappserver/Model/Node.pm for an example.